### ISSUE TITLE
Debugging [Authorize]

### ISSUE THREAD

**Issue by mstrong64** [IdentityServer/Thinktecture.IdentityServer3#832]
I have a strange situation where a webapi running in an Azure web role is unable to authenticate with idSrv running in another web role. When I access the API from a client it always returns a 401. However, if run this code outside of Azure it all works fine.

So I somehow need to step through the authentication process and see where it is breaking. I've tried putting break points in the server code but they're not triggering.

Any suggestions?

**Comment by leastprivilege** [comment]
Have you already tried logging?

**Comment by mstrong64** [comment]
Nothing helpful is appearing in the Output window. This is all I get:

`A first chance exception of type 'System.Net.Sockets.SocketException' occurred in System.dll`
`A first chance exception of type 'System.Security.Authentication.AuthenticationException' occurred in System.dll`

**Comment by mstrong64** [comment]
The additional info on the exception is:

`No connection could be made because the target machine actively refused it`

Could this be happening before it gets to idSrv? I'm using a self-signed certificate, so I don't know if that's causing a problem with the Authorize pipeline...

**Comment by brockallen** [comment]
Your one azure instance is trying to do direct connections to the other and i'm guessing your firewalls are preventing it.

**Comment by mstrong64** [comment]
Ah, yes, I didn't think about that. I will investigate. Thanks!

**Comment by mstrong64** [comment]
Turns out it was a certificate issue because the test cert is self-signed. I fixed this by including this line of code in startup:

`ServicePointManager.ServerCertificateValidationCallback += (sender, certificate, chain, sslPolicyErrors) => { return true; };`

Incidentally, I did try to get Azure internal endpoints working between the two roles, but without any success. If this could be done it would be a better solution but I don't have time to chase this now.
